src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php

```diff
@@ -157,4 +157,23 @@ public function urlRedirectAction(Request $request, string $path, bool $permanen
 
         return new RedirectResponse($url, $statusCode);
     }
+
+    public function __invoke(Request $request): Response
+    {
+        $routeParams = $request->attributes->get('_route_params', []);
+
+        if (\array_key_exists('route', $routeParams)) {
+            $parameters = array_merge(['route' => '', 'permanent' => false, 'ignoreAttributes' => false, 'keepRequestMethod' => false, 'keepQueryParams' => false], $routeParams);
+
+            return $this->redirectAction($request, ...array_values($parameters));
```

Comment:
if the route params has other route defaults that do not exist on `redirectAction`, the method will be called with non-existing params. passing params explicitly seems more sensible to me.

Reply:
Updated.